rclcpp/include/rclcpp/service.hpp

```diff
@@ -116,11 +119,22 @@ class Service : public ServiceBase
     auto service_type_support_handle = get_service_type_support_handle<ServiceT>();
 
     // rcl does the static memory allocation here
-    service_handle_ = new rcl_service_t;
-    *service_handle_ = rcl_get_zero_initialized_service();
+    service_handle_ = std::shared_ptr<rcl_service_t>(
+      new rcl_service_t, [=](rcl_service_t *service)
+      {
+        if (rcl_service_fini(service, node_handle_.get()) != RCL_RET_OK) {
+          RCUTILS_LOG_ERROR_NAMED(
```

Comment:
You should use the RCLCPP version of these macros now, see:

http://docs.ros2.org/ardent/api/rclcpp/logging_8hpp.html

This also removes the need for (and therefore implicitly address several other style comments I made) the new rcutils includes.

Reply:
Looking into it now...